src/c++/perf_analyzer/main.cc

```diff
@@ -1053,6 +1151,105 @@ main(int argc, char** argv)
         }
         break;
       }
+      case 31: {
+        if (pa::IsFile(optarg)) {
+          ssl_grpc_root_certifications_file = optarg;
+        } else {
+          Usage(
+              argv,
+              "--ssl-grpc-root-certifications-file must be a valid file path");
+        }
+        break;
+      }
+      case 32: {
+        if (pa::IsFile(optarg)) {
+          ssl_grpc_private_key_file = optarg;
+        } else {
+          Usage(argv, "--ssl-grpc-private-key-file must be a valid file path");
+        }
+        break;
+      }
+      case 33: {
+        if (pa::IsFile(optarg)) {
+          ssl_grpc_certificate_chain_file = optarg;
+        } else {
+          Usage(
+              argv,
+              "--ssl-grpc-certificate-chain-file must be a valid file path");
+        }
+        break;
+      }
+      case 34: {
+        if (std::atol(optarg) == 0 || std::atol(optarg) == 1) {
+          ssl_http_verify_peer = std::atol(optarg);
+        } else {
+          Usage(argv, "--ssl-http-verify-peer must be 0 or 1");
+        }
+        break;
+      }
+      case 35: {
+        if (std::atol(optarg) == 0 || std::atol(optarg) == 1 ||
+            std::atol(optarg) == 2) {
+          ssl_http_verify_host = std::atol(optarg);
+        } else {
+          Usage(argv, "--ssl-http-verify-host must be 0, 1, or 2");
+        }
+        break;
+      }
+      case 36: {
+        if (pa::IsFile(optarg)) {
+          ssl_http_ca_certificates_file = optarg;
+        } else {
+          Usage(
+              argv,
+              "--ssl-http-ca-certificates-file must be a valid file path");
+        }
+        break;
+      }
+      case 37: {
+        if (pa::IsFile(optarg)) {
+          ssl_http_client_certificate_file = optarg;
+        } else {
+          Usage(
+              argv,
+              "--ssl-http-client-certificate-file must be a valid file path");
+        }
+        break;
+      }
+      case 38: {
+        if (std::string(optarg) == "PEM") {
+          ssl_http_client_certificate_type =
+              triton::client::HttpSslOptions::CERTTYPE::CERT_PEM;
+        } else if (std::string(optarg) == "DER") {
+          ssl_http_client_certificate_type =
+              triton::client::HttpSslOptions::CERTTYPE::CERT_DER;
+        } else {
+          Usage(
+              argv,
+              "--ssl-http-client-certificate-type must be 'PEM' or 'DER'");
+        }
+        break;
+      }
+      case 39: {
+        if (pa::IsFile(optarg)) {
+          ssl_http_private_key_file = optarg;
+        } else {
+          Usage(argv, "--ssl-http-private-key-file must be a valid file path");
+        }
+        break;
+      }
+      case 40: {
+        if (std::string(optarg) == "PEM") {
+          ssl_http_private_key_type =
```

Comment:
As described earlier, this parsing can move to TritonCLientBAckend to avoid breaking the abstraction.

Reply:
i've left the input validation (checking the string value is from valid set) here but will put the converting-string-to-enum stuff in the triton client backend area 👍